lib/xandra/connection/utils.ex

```diff
@@ -49,11 +49,31 @@ defmodule Xandra.Connection.Utils do
     # receive the response to this frame because if we said we want to use
     # compression, this response is already compressed.
     with :ok <- :gen_tcp.send(socket, payload),
-         {:ok, %Frame{body: <<>>}} <- recv_frame(socket, compressor) do
-       :ok
+         {:ok, frame} <- recv_frame(socket, compressor) do
+      case frame do
+        %Frame{body: <<>>} ->
+          :ok
+        %Frame{kind: :authenticate} ->
```

Comment:
We should handle all possible frames here. We should add another clause that raises an error such as

```elixir
case frame do
  your stuff -> ...
  other -> raise "protocol violation, got frame: #{inspect(frame)}"
end
```